src/test/unit_tests/test_bucketspace_versioning.js

```diff
@@ -283,11 +306,25 @@ mocha.describe('bucketspace namespace_fs - versioning', function() {
             mocha.it('put object 2nd time - versioning enabled - nested', async function() {
                 const prev_version_id = await stat_and_get_version_id(full_path, nested_key1);
                 const res = await s3_uid6.putObject({ Bucket: bucket_name, Key: nested_key1, Body: body2 });
+                const version_path_nested = path.join(full_path, dir1, HIDDEN_VERSIONS_PATH);
+                const nested_versions_dir_exist = await fs_utils.file_exists(version_path_nested);
+                assert.ok(nested_versions_dir_exist);
                 const comp_res = await compare_version_ids(full_path, nested_key1, res.VersionId, prev_version_id);
                 assert.ok(comp_res);
                 const exist = await version_file_exists(full_path, key1, dir1, prev_version_id);
                 assert.ok(exist);
             });
+
+            mocha.it('put object - versioning enabled - nested key (more than 1 level)', async function() {
+                await s3_uid6.putBucketVersioning({ Bucket: nested_keys_bucket_name, VersioningConfiguration: { MFADelete: 'Disabled', Status: 'Enabled' } });
+                await s3_uid6.putObject({ Bucket: nested_keys_bucket_name, Key: nested_key_level3, Body: body1 });
+                // only after the second PUT object we expect to have the .versions under the parent directory of the file
+                await s3_uid6.putObject({ Bucket: nested_keys_bucket_name, Key: nested_key_level3, Body: body2});
+                await s3_uid6.putObject({ Bucket: nested_keys_bucket_name, Key: nested_key_level3, Body: body3});
+                const version_path_nested = path.join(nested_keys_full_path, dir_path_nested, HIDDEN_VERSIONS_PATH);
+                const exist = await fs_utils.file_exists(version_path_nested);
```

Comment:
I see this check is repeating on all tests, is this the only thing we want to check here?

Reply:
I think it's better to check that the version ids resulted from the putObject responses are really there, and not only having .versions/ dir in the path.

Reply:
I added a check on the version names in the .version/ directory to match the version id in the putObject response.